octobot_trading/exchanges/traders/trader.py

```diff
@@ -459,6 +459,29 @@ async def _handle_order_cancel_error(self, order, err, wait_for_cancelling, canc
                 f"Can't cancel order and unknown post sync order state for order: {order}."
             ) from err
 
+    async def cancel_all_orders(
+        self, symbol, wait_for_cancelling: bool = True,
+        cancelling_timeout: float = octobot_trading.constants.INDIVIDUAL_ORDER_SYNC_TIMEOUT
+    ):
+        orders_to_cancel = self.exchange_manager.exchange_personal_data.orders_manager.get_all_orders(symbol)
+        try:
+            cancel_on_exchange = False
+            for order in orders_to_cancel:
+                await order.lock.acquire()
+                if self.simulate or order.is_self_managed():
+                    order.status = octobot_trading.enums.OrderStatus.CANCELED
+                else:
+                    cancel_on_exchange = True
+            if cancel_on_exchange:
+                await self.exchange_manager.exchange.cancel_all_orders(symbol)
```

Comment:
Should we add a fallback mechanism if cancel all orders is not supported ?

Reply:
good question. I don't really know if we should as the fallback would be to cancel all orders one by one and i'm not sure we want that when calling this method.

Maybe we can add a param to allow fallback or not

Reply:
Yes we could add a param to allow fallback. If it had been supported on most exchanges, we could perhaps have left it as is, but it doesn't seem to be the case and so if there is no fallback it is not very logical compared to our framework which abstracts the specificities of exchanges.

Reply:
I'll add it 👍

Reply:
up